electron/src/js/lib/SingleSignOn.ts

```diff
@@ -0,0 +1,358 @@
+    windowOriginUrl: string,
+    private readonly windowOptions: Electron.BrowserWindowConstructorOptions
+  ) {
+    this.mainSession = this.mainBrowserWindow.webContents.session;
+    this.senderWebContents = senderEvent.sender;
+    this.windowOriginUrl = new URL(windowOriginUrl);
+  }
+
+  public readonly init = async (): Promise<void> => {
+    // Create a ephemeral and isolated session
+    this.session = session.fromPartition(SingleSignOn.SSO_SESSION_NAME, {cache: false});
+
+    // Disable browser permissions (microphone, camera...)
+    this.session.setPermissionRequestHandler((webContents, permission, callback) => callback(false));
+
+    // User-agent normalization
+    this.session.webRequest.onBeforeSendHeaders(
+      {
+        urls: ['*'],
+      },
+      (details: any, callback: Function) => {
+        details.requestHeaders['User-Agent'] = SingleSignOn.SSO_USER_AGENT;
+        callback({cancel: false, requestHeaders: details.requestHeaders});
+      }
+    );
+
+    const SingleSignOnLoginWindow = this.createBrowserWindow();
+
+    // Register protocol
+    // Note: we need to create the window before otherwise it does not work
+    await SingleSignOn.protocol.register(this.session, (type: string) => this.finalizeLogin(type));
+
+    // Show the window(s)
+    SingleSignOnLoginWindow.loadURL(this.windowOriginUrl.toString());
+
+    if (argv.devtools) {
+      SingleSignOnLoginWindow.webContents.openDevTools({mode: 'detach'});
+    }
+  };
+
+  private readonly createBrowserWindow = (): Electron.BrowserWindow => {
+    // Discard old preload URL
+    delete (<any>this.windowOptions).webPreferences.preloadURL;
+
+    const SingleSignOnLoginWindow = new BrowserWindow({
+      ...this.windowOptions,
+      alwaysOnTop: true,
+      backgroundColor: '#FFFFFF',
+      fullscreen: false,
+      fullscreenable: false,
+      height: this.windowOptions.height || 600,
+      maximizable: false,
+      minimizable: false,
+      modal: false,
+      movable: false,
+      parent: this.mainBrowserWindow,
+      resizable: false,
+      title: this.windowOriginUrl.origin,
+      titleBarStyle: 'default',
+      useContentSize: true,
+      webPreferences: {
+        ...this.windowOptions.webPreferences,
+        allowRunningInsecureContent: false,
+        backgroundThrottling: false,
+        contextIsolation: true,
+        devTools: true,
+        disableBlinkFeatures: '',
+        enableBlinkFeatures: '',
+        experimentalCanvasFeatures: false,
+        experimentalFeatures: false,
+        images: true,
+        javascript: true,
+        nativeWindowOpen: false,
+        nodeIntegration: false,
+        nodeIntegrationInWorker: false,
+        offscreen: false,
+        partition: '',
+        plugins: false,
+        preload: SingleSignOn.PRELOAD_SSO_JS,
+        sandbox: true,
+        scrollBounce: true,
+        session: this.session,
+        textAreasAreResizable: false,
+        webSecurity: true,
```

Comment:
Default is already `true`.

Reply:
But maybe not in `this.windowOptions.webPreferences`, just in case we want to make sure secure defaults are used.